core/eolearn/tests/test_utils/test_parsing.py

```diff
@@ -0,0 +1,262 @@
+            (FeatureType.DATA, "BBOX", "BBOX"),
+        ],
+        features=[
+            (FeatureType.DATA, "IS_VALID"),
+            (FeatureType.DATA, "CLM"),
+            (FeatureType.DATA, "BBOX"),
+        ],
+        renaming=[
+            (FeatureType.DATA, "IS_VALID", "new_IS_VALID"),
+            (FeatureType.DATA, "CLM", "new_CLM"),
+            (FeatureType.DATA, "BBOX", "BBOX"),
+        ],
+        specifications=[
+            (FeatureType.DATA, "IS_VALID"),
+            (FeatureType.DATA, "CLM"),
+            (FeatureType.DATA, "BBOX"),
+        ],
+    ),
+]
+
+
+TEST_CASES_ELLIPSIS = [
+    TestClass(
+        input=({FeatureType.DATA: ..., FeatureType.MASK: [("bands", "new_bands"), ("CLP", "new_CLP")]}),
+        features=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "IS_VALID"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.DATA, "CLM"),
+            (FeatureType.DATA, "BBOX"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+        renaming=[
+            (FeatureType.DATA, "bands", "bands"),
+            (FeatureType.DATA, "IS_VALID", "IS_VALID"),
+            (FeatureType.DATA, "CLP", "CLP"),
+            (FeatureType.DATA, "CLM", "CLM"),
+            (FeatureType.DATA, "BBOX", "BBOX"),
+            (FeatureType.MASK, "bands", "new_bands"),
+            (FeatureType.MASK, "CLP", "new_CLP"),
+        ],
+        specifications=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "IS_VALID"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.DATA, "CLM"),
+            (FeatureType.DATA, "BBOX"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+    ),
+    TestClass(
+        input=((FeatureType.DATA, ...), (FeatureType.MASK, "bands", "new_bands"), (FeatureType.MASK, "CLP", "new_CLP")),
+        features=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "IS_VALID"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.DATA, "CLM"),
+            (FeatureType.DATA, "BBOX"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+        renaming=[
+            (FeatureType.DATA, "bands", "bands"),
+            (FeatureType.DATA, "IS_VALID", "IS_VALID"),
+            (FeatureType.DATA, "CLP", "CLP"),
+            (FeatureType.DATA, "CLM", "CLM"),
+            (FeatureType.DATA, "BBOX", "BBOX"),
+            (FeatureType.MASK, "bands", "new_bands"),
+            (FeatureType.MASK, "CLP", "new_CLP"),
+        ],
+        specifications=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "IS_VALID"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.DATA, "CLM"),
+            (FeatureType.DATA, "BBOX"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+    ),
+]
+
+TEST_CASES_ALLOWED = [
```

Comment:
this doesn't need to be provided as test cases, since you have no output

Reply:
done